components/Input/Input.tsx

```diff
@@ -176,6 +190,30 @@ const StyledInput = styled('input', {
         paddingInlineEnd: 'calc($3 + 15px)',
       },
     },
+    {
+      endAdornment: true,
+      size: 'small',
+      state: 'invalid',
+      css: {
+        paddingInlineEnd: 'calc($4 + 30px)',
+      },
+    },
+    {
+      endAdornment: true,
+      size: 'medium',
+      state: 'invalid',
+      css: {
+        paddingInlineEnd: 'calc($6 + 30px)',
+      },
+    },
+    {
+      endAdornment: true,
+      size: 'large',
+      state: 'invalid',
+      css: {
+        paddingInlineEnd: 'calc($6 + 30px)',
```

Comment:
At first glance, it's difficult to understand the padding value.
It might help to add a comment, ex: 
```
calc($3 + $3 + 15px + 15px)
// size padding + adonrment margins + icon size + icon size
```
or
```
calc($6 + 30px)
// size padding + adonrment margins + icon size + icon size
```